fs/fs.go

```diff
@@ -263,13 +302,13 @@ func (fs *filesystem) MountLocal(ctx context.Context, mountpoint string, labels
 	return nil
 }
 
-func (fs *filesystem) getSociContext(ctx context.Context, imageRef, indexDigest string) (*sociContext, error) {
+func (fs *filesystem) getSociContext(ctx context.Context, imageRef, indexDigest, imageDigest string) (*sociContext, error) {
 	cAny, _ := fs.sociContexts.LoadOrStore(indexDigest, &sociContext{})
```

Comment:
This will be broken if `indexDigest` is empty. 

I had a suggestion to replace the map with `imageDigest` here: https://github.com/awslabs/soci-snapshotter/issues/44#issuecomment-1255771735

Is there a use-case where we would want an image to be pullable using 2 different indices? Do we support namespacing?

Reply:
I can move the Referrers API call to just before `LoadOrStore`.

Reply:
That means we have a referrers API call for every layer. Are we willing to pay that price?

Reply:
Currently, it is not possible for an image to be pulled multiple times using different indices. So maybe using the image manifest digest for `sociContexts` is the way to go about it.